packages/gafl-webapp-service/src/pages/terms-and-conditions/__tests__/route.spec.js

```diff
@@ -63,5 +80,80 @@ describe('terms-and-conditions > route', () => {
     })
+
+    it('test licenceTypeDisplay is called with permission and mssgs', async () => {
+      const func = () => getData(generateMockRequest({ [LICENCE_SUMMARY.page]: true, [CONTACT_SUMMARY.page]: false }))
+      await expect(func).rejects.toThrowRedirectTo(CONTACT_SUMMARY.uri)
+    })
+
+    const troutAndCoarse2Rods = 'Trout and coarse, up to 2 rods'
+    const troutAndCoarse3Rods = 'Trout and coarse, up to 3 rods'
+    const salmonAndSeaTrout = 'Salmon and sea trout'
+
+    it.each([[troutAndCoarse2Rods], [troutAndCoarse3Rods], [salmonAndSeaTrout]])(
+      'licenceTypeDisplay is called with permission and catalog',
+      async type => {
+        const permission = getMockPermission(10, type)
+        await getData(
+          generateMockRequest(
+            { [LICENCE_SUMMARY.page]: true, [CONTACT_SUMMARY.page]: true },
+            {
+              permissions: [permission]
+            }
+          )
+        )
+
+        expect(licenceTypeDisplay).toHaveBeenCalledWith(permission, catalog)
+      }
+    )
+
+    it.each([
+      ['Trout and coarse', true, '2'],
+      ['Trout and coarse', false, '3'],
+      ['Salmon and sea trout', false, '1']
+    ])('returns whether to display Trout and coarse, up to 2 rods conditions', async (type, displayType, rods) => {
+      const data = await getData(
+        generateMockRequest(
+          { [LICENCE_SUMMARY.page]: true, [CONTACT_SUMMARY.page]: true },
+          {
+            permissions: [getMockPermission(1, type, rods), getMockPermission(1, type, rods), getMockPermission(0, type, 3)]
+          }
+        )
+      )
+      expect(data.troutAndCoarse2Rods).toBe(displayType)
+    })
+
+    it.each([
+      ['Trout and coarse', false, '2'],
+      ['Trout and coarse', true, '3'],
+      ['Salmon and sea trout', false, '1']
+    ])('returns whether to display Trout and coarse, up to 3 rods conditions', async (type, displayType, rods) => {
+      const data = await getData(
+        generateMockRequest(
+          { [LICENCE_SUMMARY.page]: true, [CONTACT_SUMMARY.page]: true },
+          {
+            permissions: [getMockPermission(1, type, rods), getMockPermission(1, type, rods), getMockPermission(0, type, '2')]
+          }
+        )
+      )
+      expect(data.troutAndCoarse3Rods).toBe(displayType)
+    })
+
+    it.each([
+      ['Trout and coarse', false, '2'],
+      ['Trout and coarse', false, '3'],
+      ['Salmon and sea trout', true, '1']
+    ])('returns whether to display Salmon and sea trout conditions', async (type, displayType, rods) => {
+      const data = await getData(
+        generateMockRequest(
+          { [LICENCE_SUMMARY.page]: true, [CONTACT_SUMMARY.page]: true },
+          {
+            permissions: [getMockPermission(1, type, rods), getMockPermission(1, type, rods), getMockPermission(0, type, '2')]
+          }
+        )
+      )
+
+      expect(data.salmonAndSeaTrout).toBe(displayType)
+    })
```

Comment:
I've made change but unsure if done setup correctly, using let to then run async to be able get value of data, is a better way of doing this?

Reply:
ok, I see why you think you need to set the LICENCE_TYPE to be what it is - it's because you're hard-coding the strings here ('Trout and coarse' and 'Salmon and sea trout'). In fact, you should be using `LICENCE_TYPE['trout-and-coarse']` and `LICENCE_TYPE['salmon-and-sea-trout']`.

I'm not a huge fan of how you're calling `getData`, I feel like it obscures the logic a bit. If you were to create a new method called `getSamplePermissions` that will return three permissions with the given price, type and rod count. Also, you always seem to be specifying the same values for the status - so I think these should be the defaults for this function, then you can omit the arguments for the `statusGet` parameter.